pkg/platform/platform.go

```diff
@@ -298,3 +303,44 @@ func IsUserDefinedIPsecMachineConfig(machineConfig *mcfgv1.MachineConfig) bool {
 	}
 	return isSubset(machineConfig.Annotations, UserDefinedIPsecMachineConfigAnnotation)
 }
+
+// ContainsNetworkOwnerRef returns true if any one the given OwnerReference is owned
+// by cluster network operator, otherwise returns false
+func ContainsNetworkOwnerRef(ownerRefs []metav1.OwnerReference) bool {
+	for _, ownerRef := range ownerRefs {
+		if ownerRef.APIVersion == operv1.GroupVersion.String() && ownerRef.Kind == "Network" &&
+			(ownerRef.Controller != nil && *ownerRef.Controller) && ownerRef.Name == "cluster" {
+			return true
+		}
+	}
+	return false
+}
+
+// AreMachineConfigsRenderedOnPool returns true if machineConfigs are completely rendered on the given machine config
+// pool status, otherwise returns false.
+func AreMachineConfigsRenderedOnPool(status mcfgv1.MachineConfigPoolStatus, machineConfigs sets.Set[string]) bool {
+	checkSource := func(sourceNames sets.Set[string], machineConfigs sets.Set[string]) bool {
+		return sourceNames.IsSuperset(machineConfigs)
+	}
+	return status.MachineCount == status.UpdatedMachineCount &&
+		checkSourceInMachineConfigStatus(status, machineConfigs, checkSource)
+}
+
+// AreMachineConfigsRemovedFromPool returns true if machineConfigs are completely removed on the given machine config
+// pool status, otherwise returns false.
+func AreMachineConfigsRemovedFromPool(status mcfgv1.MachineConfigPoolStatus, machineConfigs sets.Set[string]) bool {
+	checkSource := func(sourceNames sets.Set[string], machineConfigs sets.Set[string]) bool {
+		return !sourceNames.HasAny(machineConfigs.UnsortedList()...)
+	}
+	return status.MachineCount == status.UpdatedMachineCount &&
+		checkSourceInMachineConfigStatus(status, machineConfigs, checkSource)
+}
+
+func checkSourceInMachineConfigStatus(machineConfigStatus mcfgv1.MachineConfigPoolStatus, machineConfigs sets.Set[string],
+	test func(sets.Set[string], sets.Set[string]) bool) bool {
+	sourceNames := sets.New[string]()
+	for _, source := range machineConfigStatus.Configuration.Source {
+		sourceNames.Insert(source.Name)
+	}
+	return test(sourceNames, machineConfigs)
+}
```

Comment:
I guess another option would have been to place these somewhere in the util package